### DISCUSSION TITLE
stepinfo

### DISCUSSION THREAD

**Discussion by bsrakd** [python-control/python-control#748] [Q&A] [answered]
According to python-control/control/timeresp.py step_info:

```python
sys = TransferFunction([-1, 1], [1, 1, 1])
S = step_info(sys)
for k in S:
    print(f"{k}: {S[k]:3.4}")
````

How is `S[k]` in the for loop determined?
For example, what would `S[k]` be if my system was `sys = TransferFunction(2, [1, 1, 2])`
or how else can I get stepinfo information from the system for python

**Comment by murrayrm** [comment, marked as answer]
The `step_info` command returns a Python dictionary object with the properties of the step response.  The `for` loop in the example iterates over the keys of the dictionary and prints the name of the key and its value.

For your example, it turns out there is a small twist: the undershoot is zero and so the format string above will generate an error due to the way that `3.4` is interpreted (this format string only works for floats).  But you can retrieve the values just by looking at the key-value pairs in `S`.  Using ipython, I get:
```
{'RiseTime': 0.9768542818762616,
 'SettlingTime': 7.814834255010094,
 'SettlingMin': 0.9069710996980537,
 'SettlingMax': 1.305008244865823,
 'Overshoot': 30.50082448658229,
 'Undershoot': 0,
 'Peak': 1.305008244865823,
 'PeakTime': 2.37236039884235,
 'SteadyStateValue': 1.0}
```

**Reply by bsrakd** [reply]
thank you for your answer 🙏
